rfcs/reftest_simplification.md

```diff
@@ -0,0 +1,56 @@
+# RFC #15: Reftest graph simplification
+
+## Summary
+
+Simplify what we allow within the reftest graph.
+
+## Details
+
+### Background
+
+Currently, reftests form a graph which can quickly end up with a large number of pass conditions.
+
+In rough principle, we load everything with `<link rel=match>`/`<link rel=mismatch>` into a directed graph (with URLs as vertices, `match`/`mismatch` as edges), and then make every source node (i.e., a node with no incoming edges) into a "test".
+
+From each test, we enumerate all possible walks from the source node with either no repeating nodes and ending at a sink node (i.e., a node with no outgoing edges), or ending at the first repeated node. Each walk becomes a sequence of ANDs, and at least one walk must be satisfied for the test to pass.
+
+In reality, we have very little complexity within the graph, as of [a12f37b6a0](https://github.com/web-platform-tests/wpt/tree/a12f37b6a04bdbde0902b0fc36b6a46f4782bd06): we have 15905 tests, and only 253 are more complex than a simple two page comparison.
+
+Of those 253:
+ * 118 just have alternates at the top level (the basic OR case),
+ * 132 are just trails (the basic AND case),
+ * leaving us with 3 more complex cases (which mix AND and OR).
+ 
+However, notably our definition of how we define the pass condition differs from what the [CSS WG had documented for their testsuite](https://wiki.csswg.org/test/format#reference-links) originally defined (in 2011), which the WPT behaviour was originally meant to match.
+
+This behaviour is that if there is at least one `<link rel=match>` at least one must pass, and all `<link rel=mismatch>` must pass.
+
+Of the 118 tests we detect as OR under our current logic, but [85 of these](https://gist.github.com/gsnedders/2ee57070569e177d973a6736f7d278bb#file-only_or_mixed_eq_cond-txt) have a combination of both match and mismatch, which is likely a bug under our current logic, and looking at a mostly random subset of these, it appears that the intention was the CSS WG documented behaviour.
```

Comment:
If the first "but" is used the sense of "only" then simpler English would be better. Otherwise there's some typo or grammar problem here.

Reply:
s/but// I think?

Reply:
The resulting sentence would make sense, yes.